File: JAVA/2109. Adding Spaces to a String.java
class Solution {
    public String addSpaces(String s, int[] d) {
        StringBuilder k=new StringBuilder();
        int j=0;
        for(int i:d){
            k.append(s.substring(j,i)).append(' ');
            j=i;
        }
        k.append(s.substring(j,s.length()));
        return k.toString();
    }
}

File: C++/2109. Adding Spaces to a String.cpp
class Solution {
public:
    string addSpaces(string s, vector<int>& d) {
        string k="";
        int j=0;
        for(int i:d){
            k+=s.substr(j,i-j)+' ';
            j=i;
        }
        k+=s.substr(j);
        return k;
    }
};